cognite/client/data_classes/filters.py

```diff
@@ -238,9 +238,12 @@ def _validate_filter(
 ) -> None:
     if filter is None or isinstance(filter, dict):
         return
-    if not_supported := (filter._involved_filter_types() - supported_filters):
-        names = [f.__name__ for f in not_supported]
-        raise ValueError(f"The filters {names} are not supported for {api_name}")
+    elif isinstance(filter, Filter):
+        if not_supported := (filter._involved_filter_types() - supported_filters):
+            names = [f.__name__ for f in not_supported]
+            raise ValueError(f"The filters {names} are not supported for {api_name}")
+    else:
+        raise ValueError(f"Invalid filter type: {type(filter)}. Must be Filter or dict")
```

Comment:
This only raises if the top level filter is incorrect. Using e.g. `filter=And(MatchAll(), set())`, still gives the same error:
`AttributeError: 'set' object has no attribute '_involved_filter_types'`

Reply:
I think that is less likely user error to do, but added a check for that as well.

Reply:
Nice